snmp/backupninja.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+import io
+import re
+import gzip
+import os
+import json
+
+LOGFILES = [
+    "/var/log/backupninja.log",
+]
+
+def main():
+    last_actions, last_fatal, last_error, last_warning = get_backupninja_state()
+    output = {'version': '1', 'error': '0', 'error_string': '', 'actions': last_actions, 'fatal': last_fatal, 'error': last_error, 'warning': last_warning}
```

Comment:
ok you missunderstood me:
json to be transfered has in general structure:
version   (versionnumber of running agent, starts with 1)
error  (error code from agent, 0 -> no error)
error_string (Errormessage, '' if no error)
data (list or dictionary with your data in here)

e.G.
```
{
version: 1,
error:0
errorstring: '',
data: {'a': 1, 'b: 2 ; 'sizes': ['kevin: 'big', 'stuart': 'medium', 'bob': 'small'], ...}
}
```

in data you're free to build your structure as you need it to transfer it to poller :)

Reply:
Thanks for the explanation !
I updated the code :-)